<?xml version="1.0"?>
<launch>
    <catch2_launch_file description="Test turtle_control node" />
    <arg name="test_duration" default="4.0" />

    <!-- <node pkg="nuturtle_control" exec="turtle_control" name="turtle_control" /> -->

    <node pkg="robot_state_publisher" exec="robot_state_publisher">
        <param name="robot_description"
            value="$(command '$(exec-in-pkg xacro xacro) $(find-pkg-share nuturtle_description)/urdf/turtlebot3_burger.urdf.xacro')" />
    </node>

    <node pkg="joint_state_publisher" exec="joint_state_publisher" />

    <node pkg="nuturtle_control" exec="odometry">
        <param name="body_id" value="base_footprint" />
        <param name="odom_id" value="odom" />
        <param name="wheel_left" value="wheel_left_joint" />
        <param name="wheel_right" value="wheel_right_joint" />
    </node>

    <catch2_integration_test_node
        pkg="nuturtle_control"
        exec="turtle_odom_test"
    >
        <param name="test_duration" value="$(var test_duration)" />
    </catch2_integration_test_node>
</launch>